sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,64 +31,146 @@
-
-    if not (isinstance(numeric_only, types.Omitted) or numeric_only is False):
-        raise TypingError("{} 'numeric_only' unsupported. Given: {}".format(_func_name, axis))
-
-    def sdc_pandas_dataframe_count_impl(self, axis=0, level=None, numeric_only=False):
-        result_data = []
-        result_index = []
-
-        for dataframe_item in self._data:
-            item_count = dataframe_item.count()
-            item_name = dataframe_item._name
-            result_data.append(item_count)
-            result_index.append(item_name)
-
-        return pandas.Series(data=result_data, index=result_index)
-
-    return sdc_pandas_dataframe_count_impl
+if not sdc.config.use_default_dataframe:
+    from sdc.datatypes.hpat_pandas_dataframe_types import DataFrameType
+
+    @overload_method(DataFrameType, 'count')
+    def sdc_pandas_dataframe_count(self, axis=0, level=None, numeric_only=False):
+        """
+        Pandas DataFrame method :meth:`pandas.DataFrame.count` implementation.
+        .. only:: developer
+            Test: python -m sdc.runtests sdc.tests.test_dataframe.TestDataFrame.test_count
+        Parameters
+        -----------
+        self: :class:`pandas.DataFrame`
+            input arg
+        axis:
+            *unsupported*
+        level:
+            *unsupported*
+        numeric_only:
+            *unsupported*
+        Returns
+        -------
+        :obj:`pandas.Series` or `pandas.DataFrame`
+                returns: For each column/row the number of non-NA/null entries.\
+                     If level is specified returns a DataFrame.
+        """
+
+        _func_name = 'Method pandas.dataframe.count().'
+
+        if not isinstance(self, DataFrameType):
+            raise TypingError('{} The object must be a pandas.dataframe. Given: {}'.format(_func_name, self))
+
+        if not (isinstance(axis, types.Omitted) or axis == 0):
+            raise TypingError("{} 'axis' unsupported. Given: {}".format(_func_name, axis))
+
+        if not (isinstance(level, types.Omitted) or level is None):
+            raise TypingError("{} 'level' unsupported. Given: {}".format(_func_name, axis))
+
+        if not (isinstance(numeric_only, types.Omitted) or numeric_only is False):
+            raise TypingError("{} 'numeric_only' unsupported. Given: {}".format(_func_name, axis))
+
+        def sdc_pandas_dataframe_count_impl(self, axis=0, level=None, numeric_only=False):
+            result_data = []
+            result_index = []
+
+            for dataframe_item in self._data:
+                item_count = dataframe_item.count()
+                item_name = dataframe_item._name
+                result_data.append(item_count)
+                result_index.append(item_name)
+
+            return pandas.Series(data=result_data, index=result_index)
+
+        return sdc_pandas_dataframe_count_impl
+
+else:
+    def sdc_pandas_dataframe_reduce_columns(df, name, param):
```

Comment:
Why you need this extra function? Could you please put this realization into `@overload_method(DataFrameType, 'head')`?

Reply:
We do this to prevent code duplication for reduce dataframe column,
 the general function for reducing the columns of a data frame is described here:
https://github.com/IntelPython/sdc/blob/eeed32775959c8d7c7cd2a3b58c55ba215934d3d/sdc/datatypes/hpat_pandas_dataframe_functions.py#L103